pyproject.toml

```diff
@@ -179,3 +179,10 @@ ignore = [
   "E701", # (multiple statements on one line)
   "S101", # (use of assert): mostly we use assert for typing
 ]
+
+[tool.ruff.format]
+quote-style = "double"
+indent-style = "space"
+ignore = [
+  "E303",  # (too many blank lines): sometimes we use multiple blank lines for separating sections
+]
```

Comment:
Many(about 60 files) changes are necessary to enable [E303](https://docs.astral.sh/ruff/rules/too-many-blank-lines/).
My suggestion is to ignore this formatting to avoid polluting git log.